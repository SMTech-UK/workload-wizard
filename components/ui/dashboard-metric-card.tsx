import { ReactNode } from "react";

interface DashboardMetricCardProps {
  title: string;
  value: ReactNode;
  subtitle?: string;
  icon?: ReactNode;
  valueClassName?: string;
  children?: ReactNode; // For custom content like Progress bar
}

export function DashboardMetricCard({
  title,
  value,
  subtitle,
  icon,
  valueClassName = "",
  children,
}: DashboardMetricCardProps) {
  return (
    <div className="rounded-lg border p-6 bg-white shadow-sm">
      <div className="flex flex-row items-center justify-between space-y-0 pb-2">
        <span className="text-sm font-medium text-gray-500">{title}</span>
        {icon}
      </div>
      <div className={`text-2xl font-bold ${valueClassName}`}>{value}</div>
      {children}
      {subtitle && <div className="text-xs text-muted-foreground mt-1">{subtitle}</div>}
    </div>
  );
} 