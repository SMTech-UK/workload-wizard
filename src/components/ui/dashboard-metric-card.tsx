import { ReactNode } from "react";

interface DashboardMetricCardProps {
  title: string;
  value: ReactNode;
  subtitle?: string;
  icon?: ReactNode;
  valueClassName?: string;
  children?: ReactNode; // For custom content like Progress bar
}

export function DashboardMetricCard({
  title,
  value,
  subtitle,
  icon,
  valueClassName = "",
  children,
}: DashboardMetricCardProps) {
  return (
    <div className="rounded-lg border p-6 bg-white dark:bg-zinc-900 shadow-sm">
      <div className="flex flex-row items-center justify-between space-y-0 pb-2">
        <span className="text-sm font-medium text-gray-900 dark:text-white">{title}</span>
        {icon}
      </div>
      <div className={`text-2xl font-bold ${valueClassName}`}>{value}</div>
      {children}
      {subtitle && <div className="text-xs text-gray-600 dark:text-gray-300 mt-1">{subtitle}</div>}
    </div>
  );
} 