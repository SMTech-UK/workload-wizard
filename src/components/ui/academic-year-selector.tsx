"use client"

import React from "react";
import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from "@/components/ui/select";
import { Badge } from "@/components/ui/badge";
import { useAcademicYear } from "@/hooks/useAcademicYear";
import { Calendar } from "lucide-react";

interface AcademicYearSelectorProps {
  className?: string;
  showActiveBadge?: boolean;
  onAcademicYearChange?: (academicYearId: string) => void;
}

export function AcademicYearSelector({ 
  className, 
  showActiveBadge = true,
  onAcademicYearChange 
}: AcademicYearSelectorProps) {
  const { 
    academicYears, 
    currentAcademicYear, 
    currentAcademicYearId, 
    changeAcademicYear 
  } = useAcademicYear();

  const handleAcademicYearChange = async (academicYearId: string) => {
    try {
      await changeAcademicYear(academicYearId);
      onAcademicYearChange?.(academicYearId);
    } catch (error) {
      console.error('Failed to change academic year:', error);
    }
  };

  // Function to format academic year name (e.g., "2025/26" -> "AY-25/26")
  const formatAcademicYearName = (yearName: string) => {
    // Extract the last 2 digits from each year part
    const parts = yearName.split('/');
    if (parts.length === 2) {
      const startYear = parts[0].slice(-2);
      const endYear = parts[1].slice(-2);
      return `AY-${startYear}/${endYear}`;
    }
    // Fallback to original name if format is unexpected
    return yearName;
  };

  if (!academicYears || academicYears.length === 0) {
    return (
      <div className={`flex items-center gap-2 text-muted-foreground ${className}`}>
        <Calendar className="h-4 w-4" />
        <span className="text-sm">No academic years configured</span>
      </div>
    );
  }

  return (
    <div className={`flex items-center gap-2 ${className}`}>
      <Select
        value={currentAcademicYearId || ""}
        onValueChange={handleAcademicYearChange}
      >
        <SelectTrigger className="w-auto min-w-[120px] flex items-center gap-2">
          <Calendar className="h-4 w-4 text-muted-foreground" />
          <span className="font-medium">
            {currentAcademicYear ? formatAcademicYearName(currentAcademicYear.name) : "Select AY"}
          </span>
        </SelectTrigger>
        <SelectContent>
          {academicYears.map((year: any) => (
            <SelectItem key={year._id} value={year._id}>
              <div className="flex items-center gap-2">
                <Calendar className="h-4 w-4 text-muted-foreground" />
                <span>{formatAcademicYearName(year.name)}</span>
                {year.isActive && showActiveBadge && (
                  <Badge variant="default" className="text-xs">Active</Badge>
                )}
                {year.isStaging && (
                  <Badge variant="secondary" className="text-xs">Staging</Badge>
                )}
              </div>
            </SelectItem>
          ))}
        </SelectContent>
      </Select>
    </div>
  );
} 