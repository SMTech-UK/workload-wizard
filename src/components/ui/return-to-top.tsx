"use client";

import { Button } from "@/components/ui/button";
import { ChevronUp, Home } from "lucide-react";
import { useState, useEffect } from "react";
import Link from "next/link";

export function ReturnToTop() {
  const [isVisible, setIsVisible] = useState(true); // Temporarily always visible for testing

  useEffect(() => {
    const toggleVisibility = () => {
      console.log("Scroll position:", window.pageYOffset);
      if (window.pageYOffset > 300) {
        console.log("Setting visible to true");
        setIsVisible(true);
      } else {
        console.log("Setting visible to false");
        setIsVisible(false);
      }
    };

    // Call once on mount to check initial state
    toggleVisibility();
    
    window.addEventListener("scroll", toggleVisibility);
    return () => window.removeEventListener("scroll", toggleVisibility);
  }, []);

  const scrollToTop = () => {
    console.log("Scrolling to top");
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  };

  console.log("ReturnToTop component render, isVisible:", isVisible);

  return (
    <>
      {/* Return to Home button - top left */}
      <Link href="/">
        <Button
          size="sm"
          variant="outline"
          className="fixed top-4 left-4 z-50 shadow-lg hover:shadow-xl transition-all duration-200"
          aria-label="Return to home"
        >
          <Home className="h-4 w-4 mr-2" />
          Home
        </Button>
      </Link>

      {/* Return to Top button - bottom left */}
      <Button
        onClick={scrollToTop}
        size="sm"
        className="fixed bottom-4 left-4 z-50 rounded-full shadow-lg hover:shadow-xl transition-all duration-200 bg-blue-500 text-white"
        aria-label="Return to top"
      >
        <ChevronUp className="h-4 w-4" />
      </Button>
    </>
  );
} 