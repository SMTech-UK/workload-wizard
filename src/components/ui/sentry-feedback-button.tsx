"use client";

import { useEffect, useState } from "react";
import * as Sentry from "@sentry/nextjs";
import { Button } from "@/components/ui/button";
import { Bug, MessageCircle } from "lucide-react";
import { useModalAwareButton } from "@/hooks/useModalAwareButton";
import { createPortal } from "react-dom";

export function SentryFeedbackButton() {
  const [feedback, setFeedback] = useState<any>(null);
  const [widget, setWidget] = useState<any>(null);
  const [mounted, setMounted] = useState(false);
  const buttonRef = useModalAwareButton();

  useEffect(() => {
    setMounted(true);
    // Get the feedback instance on the client side
    setFeedback(Sentry.getFeedback());
  }, []);

  const handleFeedbackClick = (e: React.MouseEvent) => {
    e.preventDefault();
    e.stopPropagation();
    
    if (!feedback) return;

    if (widget) {
      // If widget is open, close it
      widget.removeFromDom();
      setWidget(null);
    } else {
      // Create and show the form immediately
      feedback.createForm().then((newWidget: any) => {
        newWidget.appendToDom();
        newWidget.open();
        setWidget(newWidget);
      }).catch((error: any) => {
        console.error('Failed to create Sentry feedback form:', error);
      });
    }
  };

  if (!feedback || !mounted) return null;

  const button = (
    <Button
      ref={buttonRef}
      variant="destructive"
      size="lg"
      onClick={handleFeedbackClick}
      onMouseDown={(e) => {
        e.preventDefault();
        e.stopPropagation();
      }}
      className="fixed bottom-4 right-4 z-[99999] shadow-lg"
      style={{
        zIndex: 99999,
        pointerEvents: 'auto',
      }}
      aria-label="Report a bug or give feedback"
    >
      <Bug className="h-4 w-4 mr-2" />
      Report an Issue
    </Button>
  );

  // Render the button in a portal to ensure it's outside modal context
  return createPortal(button, document.body);
} 