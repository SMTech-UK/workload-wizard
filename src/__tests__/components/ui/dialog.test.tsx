import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Dialog, DialogContent, DialogDescription, DialogFooter, DialogHeader, DialogTitle, DialogTrigger } from '@/components/ui/dialog';
import { Button } from '@/components/ui/button';

// Mock ResizeObserver
global.ResizeObserver = jest.fn().mockImplementation(() => ({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
}));

describe('Dialog Component - Comprehensive UI Testing', () => {
  const user = userEvent.setup();

  describe('Basic Rendering and Functionality', () => {
    it('should render dialog trigger button', () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
              <DialogDescription>This is a test dialog</DialogDescription>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      expect(screen.getByRole('button', { name: 'Open Dialog' })).toBeInTheDocument();
    });

    it('should open dialog when trigger is clicked', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
              <DialogDescription>This is a test dialog</DialogDescription>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByRole('dialog')).toBeInTheDocument();
        expect(screen.getByText('Test Dialog')).toBeInTheDocument();
        expect(screen.getByText('This is a test dialog')).toBeInTheDocument();
      });
    });

    it('should close dialog when escape key is pressed', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByRole('dialog')).toBeInTheDocument();
      });

      await user.keyboard('{Escape}');

      await waitFor(() => {
        expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
      });
    });

    it('should close dialog when clicking outside', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByRole('dialog')).toBeInTheDocument();
      });

      // Click outside the dialog
      const overlay = screen.getByRole('dialog').parentElement;
      if (overlay) {
        await user.click(overlay);
      }

      await waitFor(() => {
        expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
      });
    });
  });

  describe('Accessibility Testing', () => {
    it('should have proper ARIA attributes', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
              <DialogDescription>This is a test dialog</DialogDescription>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        const dialog = screen.getByRole('dialog');
        expect(dialog).toHaveAttribute('aria-modal', 'true');
        expect(dialog).toHaveAttribute('aria-labelledby');
        expect(dialog).toHaveAttribute('aria-describedby');
      });
    });

    it('should have proper focus management', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
            </DialogHeader>
            <DialogFooter>
              <Button>Cancel</Button>
              <Button>Confirm</Button>
            </DialogFooter>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByRole('dialog')).toBeInTheDocument();
      });

      // Focus should be trapped within the dialog
      const dialog = screen.getByRole('dialog');
      expect(dialog).toHaveFocus();
    });

    it('should announce dialog to screen readers', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
              <DialogDescription>This is a test dialog</DialogDescription>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        const dialog = screen.getByRole('dialog');
        const title = screen.getByText('Test Dialog');
        const description = screen.getByText('This is a test dialog');
        
        expect(dialog).toHaveAttribute('aria-labelledby', title.id);
        expect(dialog).toHaveAttribute('aria-describedby', description.id);
      });
    });

    it('should handle keyboard navigation properly', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
            </DialogHeader>
            <DialogFooter>
              <Button>Cancel</Button>
              <Button>Confirm</Button>
            </DialogFooter>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByRole('dialog')).toBeInTheDocument();
      });

      // Tab navigation should work within dialog
      await user.tab();
      expect(screen.getByRole('button', { name: 'Cancel' })).toHaveFocus();
      
      await user.tab();
      expect(screen.getByRole('button', { name: 'Confirm' })).toHaveFocus();
    });
  });

  describe('Responsive Behavior', () => {
    it('should adapt to different screen sizes', async () => {
      // Mock different screen sizes
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 768, // Tablet size
      });

      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        const dialog = screen.getByRole('dialog');
        expect(dialog).toBeInTheDocument();
        // Dialog should be responsive and adapt to screen size
        expect(dialog).toHaveClass('sm:max-w-lg');
      });
    });

    it('should handle mobile viewport correctly', async () => {
      // Mock mobile screen size
      Object.defineProperty(window, 'innerWidth', {
        writable: true,
        configurable: true,
        value: 375, // Mobile size
      });

      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        const dialog = screen.getByRole('dialog');
        expect(dialog).toBeInTheDocument();
        // Should have mobile-specific classes
        expect(dialog.parentElement).toHaveClass('fixed', 'inset-0');
      });
    });
  });

  describe('Content Structure', () => {
    it('should render dialog with header, content, and footer', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
              <DialogDescription>This is a test dialog</DialogDescription>
            </DialogHeader>
            <div>Dialog content goes here</div>
            <DialogFooter>
              <Button>Cancel</Button>
              <Button>Confirm</Button>
            </DialogFooter>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByText('Test Dialog')).toBeInTheDocument();
        expect(screen.getByText('This is a test dialog')).toBeInTheDocument();
        expect(screen.getByText('Dialog content goes here')).toBeInTheDocument();
        expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
        expect(screen.getByRole('button', { name: 'Confirm' })).toBeInTheDocument();
      });
    });

    it('should handle dialog without description', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByText('Test Dialog')).toBeInTheDocument();
        expect(screen.queryByText('This is a test dialog')).not.toBeInTheDocument();
      });
    });

    it('should handle dialog without footer', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
            </DialogHeader>
            <div>Dialog content goes here</div>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByText('Test Dialog')).toBeInTheDocument();
        expect(screen.getByText('Dialog content goes here')).toBeInTheDocument();
        expect(screen.queryByRole('button', { name: 'Cancel' })).not.toBeInTheDocument();
      });
    });
  });

  describe('User Interactions', () => {
    it('should handle multiple dialogs correctly', async () => {
      render(
        <div>
          <Dialog>
            <DialogTrigger asChild>
              <Button>Open Dialog 1</Button>
            </DialogTrigger>
            <DialogContent>
              <DialogHeader>
                <DialogTitle>Dialog 1</DialogTitle>
              </DialogHeader>
            </DialogContent>
          </Dialog>
          <Dialog>
            <DialogTrigger asChild>
              <Button>Open Dialog 2</Button>
            </DialogTrigger>
            <DialogContent>
              <DialogHeader>
                <DialogTitle>Dialog 2</DialogTitle>
              </DialogHeader>
            </DialogContent>
          </Dialog>
        </div>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog 1' }));

      await waitFor(() => {
        expect(screen.getByText('Dialog 1')).toBeInTheDocument();
      });

      await user.click(screen.getByRole('button', { name: 'Open Dialog 2' }));

      await waitFor(() => {
        expect(screen.getByText('Dialog 2')).toBeInTheDocument();
        // First dialog should be closed when second opens
        expect(screen.queryByText('Dialog 1')).not.toBeInTheDocument();
      });
    });

    it('should handle form submission in dialog', async () => {
      const handleSubmit = jest.fn();

      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Form Dialog</DialogTitle>
            </DialogHeader>
            <form onSubmit={handleSubmit}>
              <input type="text" placeholder="Enter name" />
              <DialogFooter>
                <Button type="button">Cancel</Button>
                <Button type="submit">Submit</Button>
              </DialogFooter>
            </form>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByPlaceholderText('Enter name')).toBeInTheDocument();
      });

      await user.type(screen.getByPlaceholderText('Enter name'), 'John Doe');
      await user.click(screen.getByRole('button', { name: 'Submit' }));

      expect(handleSubmit).toHaveBeenCalled();
    });
  });

  describe('Edge Cases and Error Handling', () => {
    it('should handle very long content gracefully', async () => {
      const longContent = 'A'.repeat(1000);

      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Long Content Dialog</DialogTitle>
            </DialogHeader>
            <div>{longContent}</div>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByText('Long Content Dialog')).toBeInTheDocument();
        expect(screen.getByText(longContent)).toBeInTheDocument();
      });
    });

    it('should handle special characters in content', async () => {
      const specialContent = 'Test with special chars: <>&"\'';

      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Special Characters</DialogTitle>
            </DialogHeader>
            <div>{specialContent}</div>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByText('Special Characters')).toBeInTheDocument();
        expect(screen.getByText(specialContent)).toBeInTheDocument();
      });
    });

    it('should handle empty content gracefully', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Empty Dialog</DialogTitle>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByText('Empty Dialog')).toBeInTheDocument();
        expect(screen.getByRole('dialog')).toBeInTheDocument();
      });
    });
  });

  describe('Snapshot Testing', () => {
    it('should maintain consistent UI structure for dialog components', () => {
      // Arrange & Act
      const { container: triggerContainer } = render(
        <Dialog>
          <DialogTrigger>Open Dialog</DialogTrigger>
          <DialogContent>Dialog content</DialogContent>
        </Dialog>
      );
      
      const { container: headerContainer } = render(
        <Dialog defaultOpen>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Dialog Title</DialogTitle>
              <DialogDescription>Dialog Description</DialogDescription>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );
      
      const { container: footerContainer } = render(
        <Dialog defaultOpen>
          <DialogContent>
            <DialogFooter>
              <button>Save</button>
              <button>Cancel</button>
            </DialogFooter>
          </DialogContent>
        </Dialog>
      );
      
      // Assert
      expect(triggerContainer).toMatchSnapshot('dialog-trigger-structure');
      expect(headerContainer).toMatchSnapshot('dialog-header-structure');
      expect(footerContainer).toMatchSnapshot('dialog-footer-structure');
    });

    it('should maintain consistent UI structure with complete dialog', () => {
      // Arrange & Act
      const { container } = render(
        <Dialog defaultOpen>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Complete Dialog</DialogTitle>
              <DialogDescription>This is a complete dialog example</DialogDescription>
            </DialogHeader>
            <div>Main dialog content goes here</div>
            <DialogFooter>
              <button>Save</button>
              <button>Cancel</button>
            </DialogFooter>
          </DialogContent>
        </Dialog>
      );
      
      // Assert
      expect(container).toMatchSnapshot('dialog-complete-structure');
    });

    it('should maintain consistent UI structure with custom styling', () => {
      // Arrange & Act
      const { container } = render(
        <Dialog defaultOpen>
          <DialogContent className="bg-gradient-to-br from-purple-100 to-blue-100 border-purple-300">
            <DialogHeader>
              <DialogTitle className="text-purple-800">Styled Dialog</DialogTitle>
              <DialogDescription className="text-purple-600">Custom styled dialog</DialogDescription>
            </DialogHeader>
            <div className="text-purple-700">Styled content</div>
          </DialogContent>
        </Dialog>
      );
      
      // Assert
      expect(container).toMatchSnapshot('dialog-custom-styling');
    });
  });

  describe('Performance Testing', () => {
    it('should handle rapid open/close operations', async () => {
      render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      const trigger = screen.getByRole('button', { name: 'Open Dialog' });

      // Rapid clicks
      await user.click(trigger);
      await user.click(trigger);
      await user.click(trigger);

      await waitFor(() => {
        expect(screen.getByRole('dialog')).toBeInTheDocument();
      });
    });

    it('should handle multiple dialogs efficiently', async () => {
      const { rerender } = render(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog</DialogTitle>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await user.click(screen.getByRole('button', { name: 'Open Dialog' }));

      await waitFor(() => {
        expect(screen.getByRole('dialog')).toBeInTheDocument();
      });

      // Rerender with different content
      rerender(
        <Dialog>
          <DialogTrigger asChild>
            <Button>Open Dialog 2</Button>
          </DialogTrigger>
          <DialogContent>
            <DialogHeader>
              <DialogTitle>Test Dialog 2</DialogTitle>
            </DialogHeader>
          </DialogContent>
        </Dialog>
      );

      await waitFor(() => {
        expect(screen.getByText('Test Dialog 2')).toBeInTheDocument();
      });
    });
  });
}); 