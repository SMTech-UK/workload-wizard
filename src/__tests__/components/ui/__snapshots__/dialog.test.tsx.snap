// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Component - Comprehensive UI Testing Snapshot Testing should maintain consistent UI structure for dialog components: dialog-footer-structure 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;

exports[`Dialog Component - Comprehensive UI Testing Snapshot Testing should maintain consistent UI structure for dialog components: dialog-header-structure 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;

exports[`Dialog Component - Comprehensive UI Testing Snapshot Testing should maintain consistent UI structure for dialog components: dialog-trigger-structure 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
>
  <button
    aria-controls="radix-:r1p:"
    aria-expanded="false"
    aria-haspopup="dialog"
    data-state="closed"
    type="button"
  >
    Open Dialog
  </button>
</div>
`;

exports[`Dialog Component - Comprehensive UI Testing Snapshot Testing should maintain consistent UI structure with complete dialog: dialog-complete-structure 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;

exports[`Dialog Component - Comprehensive UI Testing Snapshot Testing should maintain consistent UI structure with custom styling: dialog-custom-styling 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;
